x/supplier/keeper/msg_server_stake_supplier.go

```diff
@@ -140,15 +125,20 @@ func (k msgServer) StakeSupplier(ctx context.Context, msg *suppliertypes.MsgStak
 		return nil, status.Error(codes.InvalidArgument, err.Error())
 	}
 
-	// Send the coins from the message signer account to the staked supplier pool
 	supplierStakingFee := k.GetParams(ctx).StakingFee
-	stakeWithFee := sdk.NewCoins(coinsToEscrow.Add(*supplierStakingFee))
-	err = k.bankKeeper.SendCoinsFromAccountToModule(ctx, msgSignerAddress, suppliertypes.ModuleName, stakeWithFee)
-	if err != nil {
-		logger.Info(fmt.Sprintf("ERROR: could not send %v coins from %q to %q module account due to %v", coinsToEscrow, msgSignerAddress, suppliertypes.ModuleName, err))
-		return nil, status.Error(codes.InvalidArgument, err.Error())
+
+	if err = k.transferSupplierStakeDiff(ctx, msgSignerAddress, supplierCurrentStake, *msg.Stake); err != nil {
```

Comment:
s/transfer/handle

Reply:
Thought on "reconcile"? Handle seems even more generic to me.